Implements #54-Welcome email feature

AuraOfDivinity opened the pull request:
Implements #54 

A welcome email is now sent upon registration. Please check the screenshot below.

![Email](https://user-images.githubusercontent.com/31003923/74920674-8b26fa80-53f2-11ea-9466-442b4bbaca04.gif)

The emails are sent using the sendgrid platform. (I have kept the API key for testing purposes)

Would like to know your thoughts. @jaskirat2000

jaskirat2000 commented:
Isn't sendgrid platform paid ? Do we have any substitute which is open source and free ?

AuraOfDivinity commented:
@jaskirat2000 Sendgrid has a free tier with 30,000 emails for the first month and then 100 emails per month from next month onwards. ([Details here](https://sendgrid.com/pricing/)). I don't know of any entirely free ones though :( . Most of them have a limited free tier +paid model like SendGrid.

jaskirat2000 commented:
Not sure if third party(SendGrid) would be worth adding. I am taking into consideration about the scalability if donut grows and we can't restrict to 100 users only who signups.

@devesh-verma and @vaibhavdaren what do you think

TG1999 commented:
You can use Nodemailer it's free

AuraOfDivinity commented:
@TG1999 Thanks for pointing out. I did use nodemailer but I thought it was a must to use a 3rd party transporter like sendgrid etc and didn't consider the option of creating an SMTP server to do the same thing for free. @jaskirat2000 Would it be fine if I created a new gmail account and configure it send emails? Apologies for the misunderstanding on my part.

devesh-verma commented:
> @jaskirat2000 Sendgrid has a free tier with 30,000 emails for the first month and then 100 emails per month from next month onwards. ([Details here](https://sendgrid.com/pricing/)). I don't know of any entirely free ones though :( . Most of them have a limited free tier +paid model like SendGrid. I'm open for suggestions!

@jaskirat2000 **40,000 emails for 30 days, then 100/day forever**.
I feel it quite good for now, we can continue with SendGrid.

jaskirat2000 commented:
Oh I misunderstood 100/month but thanks I think then 100/day is quite
enough. Let's go ahead with it then.

On Fri, 21 Feb 2020, 6:16 am Devesh Verma, <notifications@github.com> wrote:

> @jaskirat2000 <https://github.com/jaskirat2000> Sendgrid has a free tier
> with 30,000 emails for the first month and then 100 emails per month from
> next month onwards. (Details here <https://sendgrid.com/pricing/>). I
> don't know of any entirely free ones though :( . Most of them have a
> limited free tier +paid model like SendGrid. I'm open for suggestions!
>
> @jaskirat2000 <https://github.com/jaskirat2000> *40,000 emails for 30
> days, then 100/day forever*.
> I feel it quite good for now, we can continue with SendGrid.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/codeuino/social-platform-donut-backend/pull/55?email_source=notifications&email_token=ACOBHKZZ5ZGBIEBZCMCPOJ3RD4P6RA5CNFSM4KYK4CC2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEMRDHAI#issuecomment-589443969>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACOBHKY4HKCT7XU2ZI4LRBTRD4P6RANCNFSM4KYK4CCQ>
> .
>

AuraOfDivinity commented:
@devesh-verma I have moved emails into a separate service. Also removed the key which was added to dev.env file which was added as a mistake. Do let me know if any other changes are required! :+1:

vaibhavdaren commented on .env.test:
```diff
@@ -1,4 +1,5 @@
-DATABASE_URL=mongodb+srv://donut-admin:5cdS2C2g3wRAdQWp@donut-users-hdawt.mongodb.net/donut-testing?retryWrites=true&w=majority
\ No newline at end of file
+DATABASE_URL=mongodb+srv://donut-admin:5cdS2C2g3wRAdQWp@donut-users-hdawt.mongodb.net/donut-testing?retryWrites=true&w=majority
+SENDGRID_API_KEY=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
```
this can be misused

AuraOfDivinity replied:
I didn't realize I have pushed the keys. I will remove them. Thanks for pointing put @vaibhavdaren

AuraOfDivinity commented:
@devesh-verma Closing this as I made a huge mess caused by rebasing on top of a set of force pushed commits :( . I will commit the changes on a new PR.
